### ISSUE TITLE
Bootstrap complexity creep due to ROUTER_PRIMARY_COMPONENT

### ISSUE THREAD

**Issue by IgorMinar** [angular/angular#4318]
With the recent platform/bootstrap refactoring we now added a requirement for people to bind the primary component during bootstrap:

```
 bootstrap(AppCmp, [
   ROUTER_BINDINGS,
   bind(ROUTER_PRIMARY_COMPONENT).toValue(AppCmp)
  ]);
```

Since 99% of applications will not utilize the multiple app feature which this api change enables, we are making the bootstrapping significantly more complicated for the majority of our developers.

Can we consider creating a special bootstrap entry point for multi-app apps, while keeping the normal boostrap simple for common use-cases?

**Comment by PascalPrecht** [comment]
Do you think of an replacement for the currently exported `routerBindings(PRIMARY_APP)` helper function?

I just had a discussion about this with @btford, where he mentioned that we might just want to export that helper function. Didn't think about multi-app apps yet, also not sure how this affects the bootstrap process.

AFAIK we use `platform()` to bootstrap multiple apps with the same change detection etc.. Does `routerBindings()` make it simpler then? Is that the solution you're looking for?

**Comment by btford** [comment]
@PascalPrecht the helper still has the same additional complexity problem.

**Comment by alxhub** [comment]
The goal is to actually get rid of `ROUTER_PRIMARY_COMPONENT` as well, and have `Router` inject and query `ApplicationRef` to retrieve the root component(s) and process their configuration.

I agree that `routerBindings()` vs `ROUTER_BINDINGS` is confusing - I didn't want to remove the latter because `routerBindings()` is only temporary until `Router` implements the above.

**Comment by mhevery** [comment]
@IgorMinar what @alxhub said https://github.com/angular/angular/issues/4318#issuecomment-142412617 I believe he is working on redoing this work.

**Comment by IgorMinar** [comment]
is there an issue for the refactoring? if not, we should make this the issue and put it on the breaking hotlist.

**Comment by mhevery** [comment]
Done

**Comment by IgorMinar** [comment]
we need this in a40. @alxhub sgty?

**Comment by IgorMinar** [comment]
fixed by #4643
